### DISCUSSION TITLE
获得的最快 IP 不能保持长期稳定使用，有没有什么方法可以解决？

### DISCUSSION THREAD

**Discussion by loofeal** [XIU2/CloudflareSpeedTest#48] [讨论]
我一直在想怎么可以实现本地与CF 的最优地址转换。不然更换次数有点太多了

**Comment by XIU2** [comment]
Cloudflare CDN 使用了 **Anycast（任播技术）**，这意味着所有的 CDN 节点都是**动态**的，随时都可能会调整。
因此 IP 质量的波动是无解的，测速获得最快 IP 后，可能是过一会就动态调整了（点背），可能过几天才会动态调整（运气好）。

**这是不可控的。**

所以我才写了个[**自动更新 Hosts 的脚本**](https://github.com/XIU2/CloudflareSpeedTest/issues/42)，至少满足我个人需求了。
毕竟我写这个软件，最初的目的就是为了**改 Hosts 加速使用 Cloudflare CDN 的网站访问速度**。

****

目前而言，只能**定期测速**来获得最新可用的**最快 IP**，至少我想不到还有什么更好的方法了，当然有什么想法可以提。

**Reply by loofeal** [reply]
是的  确实非常棒。我想如果可以找到一个桥梁  那么或许是可以实现的。中间就用来时刻的计算。得到好的地址就放入域名中绑定。找一个阈值去调整，当然仅仅是想法。

**Reply by XIU2** [reply]
但是问题在于，要得到**最真实准确的最快 IP**，就必须在**当前网络下测速**，否则你在家里测速得到的最快 IP，别人家里去用可能就很慢（也可能快，说不准）。

如果在其他网络下（服务器什么的）测速，得到的最快 IP 可能**并不适用**于你当前的网络（家庭宽带）。

**Reply by loofeal** [reply]
你说的也是。中间的容器，只有自己的属性。

**Reply by XIU2** [reply]
而如果在当前网络下，频繁的测速（比如挂在路由器上），只会导致占用大量（家庭）宽带。
这样反而得不偿失了。

****

不过一般来说，同一座城市，同一个运营商，测速获得的**最快 IP** 应该同样速度快。
当然前提条件就是，两者都是**家庭宽带**，如果是商业宽带/企业宽带，可能会速度显著提高。

> 前段时间有个 [Issues](https://github.com/XIU2/CloudflareSpeedTest/issues/21) 反馈，同一座城市，都是联通，结果公司和家里的测速结果**天差地别**。

**Reply by llkeji** [reply]
不如把程序变网站源码，让大家用网站去实现测试最适合自己宽带的IP

**Reply by XIU2** [reply]
@llkeji 想要测试获得最快的 IP，只能在本地网络下测速，如果做成一个网站，就只能用前端测速而不能是后端测速（后端就是网站服务器），前端测速意味着要用 JavaScript 等重写该软件，然而前端并不适合干这个（效率低+限制多）。

**Reply by felixding** [reply]
> @llkeji 然而前端并不适合干这个（效率低+限制多）。

用[WebAssembly](https://webassembly.org/)可行吗？
